src/com/owncloud/android/ui/fragment/OCFileListFragment.java

```diff
@@ -335,60 +343,89 @@ private void removeFabLabels() {
     }
 
     private void registerLongClickListener() {
-        getListView().setOnItemLongClickListener(new AdapterView.OnItemLongClickListener() {
-            public boolean onItemLongClick(AdapterView<?> arg0, View v,
-                                           int index, long arg3) {
-                showFileAction(index);
-                return true;
-            }
-        });
-    }
+        AbsListView listView = getListView();
+        setChoiceMode(ListView.CHOICE_MODE_MULTIPLE_MODAL);
+        setMultiChoiceModeListener(new AbsListView.MultiChoiceModeListener() {
 
+            private Menu menu;
 
-    private void showFileAction(int fileIndex) {
-        Bundle args = getArguments();
-        PopupMenu pm = new PopupMenu(getActivity(),null);
-        Menu menu = pm.getMenu();
+            @Override
+            public void onItemCheckedStateChanged(ActionMode mode, int position, long id, boolean checked) {
+                final int checkedCount = getListView().getCheckedItemCount();
+                // TODO Tobi extract to values
+                mode.setTitle(checkedCount + " selected");
 
-        boolean allowContextualActions =
-                (args == null) ? true : args.getBoolean(ARG_ALLOW_CONTEXTUAL_ACTIONS, true);
+                if (checked) {
+                    mAdapter.setNewSelection(position, checked);
+                } else {
+                    mAdapter.removeSelection(position);
+                }
 
-        if (allowContextualActions) {
-            MenuInflater inflater = getActivity().getMenuInflater();
+                OCFile targetFile = null;
+                if (checkedCount > 0 && checkedCount <= MIN_FILES_FOR_MULTISELECT) {
+                    targetFile = (checkedCount == MIN_FILES_FOR_MULTISELECT)
+                        ? mFile : mAdapter.getCheckedItems().get(0);
```

Comment:
Currently I have sent an OCFile to a FileMenuFilter to filter the actions shown in the menu: for single selection the item selected, and for multiple selection the parent folder.

Maybe update the Filter to receive all the checked items could make more sense. I leave open this discussion, and feel free to improve it :smile: